src/containers/LoginRegister/index.jsx

```diff
@@ -0,0 +1,165 @@
+import React, { useState } from "react";
+import { withRouter } from "react-router-dom";
+import { useTranslation } from "react-i18next";
+import { Alert, Spin } from "antd";
+import firebase from "firebase";
+import Login from "../../components/userauth/Login";
+import Register from "../../components/userauth/Register";
+import ResetPassword from "../../components/userauth/ResetPassword";
+import firestore, { auth } from "../../firebaseConfig";
+import "./index.css";
+import kids from "../../images/LoginKids.svg";
+
+const LoginRegisterPage = ({ history }) => {
+  const [t] = useTranslation();
+  const [active, setActive] = useState("login");
+  const [error, setError] = useState("");
+  const [message, setMessage] = useState("");
+  const [loading, setLoading] = useState(false);
+
+  const resetErrorAndMessage = () => {
+    setError("");
+    setMessage(<></>);
+  };
+  const fetchAction = async (func) => {
+    resetErrorAndMessage();
+    try {
+      setLoading(true);
+      func();
+      setLoading(() => false);
+      history.push("/");
+    } catch (e) {
+      setLoading(() => false);
+      setError(e.message);
+    }
+  };
+  const login = async (email, password) => {
+    const action = async () => {
+      await auth.signInWithEmailAndPassword(email, password);
+    };
+    fetchAction(action);
+  };
+  const passwordReset = async (email) => {
+    const action = async () => {
+      await auth.sendPasswordResetEmail(email);
+      setMessage(
+        <Alert
+          className="authAlert"
+          showIcon
+          message={
+            <div>
+              {t("loginRegister.anEmailSent")}
+              <b>{email}</b>
+              {t("loginRegister.anEmailSentContain")}
+            </div>
+          }
+          type="success"
+        />
+      );
+    };
+    fetchAction(action);
+  };
+  const loginWithFacebook = async () => {
+    const action = () => {
```

Comment:
just inline this action on line 69 and everywhere else similarly, its cleaner

Reply:
PTAL